### PR TITLE
Fix members

### PR DESCRIPTION
**Pull request by sbishel**
#### Summary
Fixes issues to display the correct team or board members when @mentioning or in a person property. 

Also introduces enum to ensure consistency when storing, converting to correct language only for display.

#### Ticket Link

  Fixes https://github.com/mattermost/focalboard/issues/3307
  Fixes https://github.com/mattermost/focalboard/issues/3864

### CODE REVIEW

File: webapp/src/components/viewHeader/filterEntry.tsx [hunk 1]
```diff
@@ -126,7 +126,7 @@ const FilterEntry = (props: Props): JSX.Element => {
                         <>
                             <Menu.Text
                                 id='is'
-                                name={intl.formatMessage({id: 'Filter.contains', defaultMessage: 'is'})}
+                                name={intl.formatMessage({id: 'Filter.is', defaultMessage: 'is'})}
```
Comment (sbishel) [inline webapp/src/components/viewHeader/filterEntry.tsx:129]: Caused issues running i18n-extract.

File: webapp/src/mutator.ts [hunk 2]
```diff
@@ -1110,7 +1108,7 @@ class Mutator {
         const boardTemplate = createBoard()
         boardTemplate.isTemplate = true
         boardTemplate.teamId = teamId
-        boardTemplate.title = intl.formatMessage({id: 'View.NewTemplateTitle', defaultMessage: 'Untitled Template'})
+        boardTemplate.title = intl.formatMessage({id: 'View.NewTemplateTitle', defaultMessage: 'Untitled'})
```
Comment (sbishel) [inline webapp/src/mutator.ts:1111]: Caused conflicts with i18n-extract

**Comment by wuwinson** [comment]
https://user-images.githubusercontent.com/93531870/191125059-c188dd12-f25b-4341-a747-6206b81845fd.mov

https://github.com/mattermost/focalboard/issues/3307 is partially fixed:
1. Create new board
2. Add person property
3. Select a non-board member (fixed)
4. Assign role to non-board member
5. Open Share modal
6. See error - user's role displays as admin, but when clicking on the role dropdown, the proper role is assigned
7. Repeat steps 2-6 with @mention and the same issue occurs

https://github.com/mattermost/focalboard/issues/3864 not fixed:
1. Create a new board
2. Open share modal and set team role to Commenter
3. Add an explicit board member
4. See error - user is still added as "Editor" rather than "Commenter"
5. Additionally, no longer able to set the board back to private.

**Comment by wuwinson** [comment]
https://github.com/mattermost/focalboard/issues/3307 is fixed, but https://github.com/mattermost/focalboard/issues/3864 is still not fixed:

1. Create a new board
2. Open share modal and set team role to "Commenter"
3. Add an explicit board member
4. See error - the user is still incorrectly added as "Editor" rather than "Commenter"
5. Remove the same explicit member from board (step 3)
6. Open a card and add the user via the person property and/or @mentioning
7. Modal appears for admin to assign a role to the board member
8. See error - Able to select "Viewer" role from the dropdown even though "Commenter" was set as the min role (step 2)
9. Open Share modal - User is added as a "Viewer" even though min role is set to "Commenter"

https://user-images.githubusercontent.com/93531870/192323545-570e4175-c0cc-492a-8d35-e55ff36f1a6e.mov

**Comment by wuwinson** [comment]
> 
Steps 1-4 are now fixed, but steps 6-9 are still an issue.

https://user-images.githubusercontent.com/93531870/192592370-9dfe0e67-9e56-48c7-89c7-11f1ccb08a61.mov

**Comment by sbishel** [comment]
I'm still working on this, but actually 6-9, I don't think are an issue. An admin should be able to add a user at any level. They can change it in share board, but why make them go there and update. I'm still working on some of the styling, but I don't think we should restrict the admin.

**Comment by wuwinson** [comment]
Got it, thanks Scott. 6-9 would create an inconsistent experience from share board, because the admin wouldn't be able to assign a role lower than the min role from share boards. It would also create a confusing experience with "min roles", otherwise we could have kept "default roles" like what we have in v7.2.1.

**Comment by wuwinson** [comment]
Steps 6-9 are still occurring, but everything else looks good @sbishel. You can't assign a role lower than the minimum role from the Share boards modal (the role is not available to select), and if you change the minimum role, it would promote everyone on the board to that role, so allowing an admin to set a lower role for an individual user only from the @mention/person property modal can be misleading.

**Comment by sbishel** [comment]
@harshilsharma63 @jespino - Can you take a look at this when you get a chance? There are a couple other bugs dependent on this fix and this one is important to 7.3 release.

**Comment by harshilsharma63** [comment]
@sbishel for #3864 adding a new user to board does follow the min role set for the team, but adding a new channel sets the channel's roles to editor. Should linked boards follow the same team-level min role? cc @wuwinson 


https://user-images.githubusercontent.com/18575143/193204527-4c61c368-1335-43f8-8ff4-72e561101d16.mp4

When adding a new board member by adding them to an assignee property, the font size for some texts in the popup seem off. I think you're already fixing it but still adding here just so its not missed

Screenshot 2022-09-30 at 11 58 00 AM

Everything else looks great!

**Review by harshilsharma63 — APPROVED** [review]
LGTM 👍

**Comment by wuwinson** [comment]
> @sbishel for #3864 adding a new user to board does follow the min role set for the team, but adding a new channel sets the channel's roles to editor. Should linked boards follow the same team-level min role? cc @wuwinson
> 
>  min-role-for-channel.mp4 
> ~When adding a new board member by adding them to an assignee property, the font size for some texts in the popup seem off. I think you're already fixing it but still adding here just so it's not missed~ I see you already have aPR for it #3887
> 
> Screenshot 2022-09-30 at 11 58 00 AM
> 
> Everything else looks great!

@harshilsharma63 Thanks for calling that out. We intentionally left out linked channels in this release due to bandwidth. We'd have to update the modal when linking a board on the RHS as well and include the new roles in the dropdown for the linked board modal. Was easier to keep linked channels on the "Editor" role for now, since they'd always be equal or above the min role set for the team.

**Pull request merged**